packages/cli/src/commands/dev.ts

```diff
@@ -142,15 +145,28 @@ export async function dev(file: string, { preact, publicDir }: { preact: boolean
     clearCache(deployment);
   });
 
-  fastify.listen(1234, 'localhost', (err, address) => {
-    if (err) {
-      console.error(err);
-      process.exit(1);
+  // get an available port to listen.
+  let retryCount = 0;
+  async function start(nextPort: number) {
+    try {
+      const address = await fastify.listen(nextPort, host);
+      console.clear();
+      logSpace();
+      logSuccess(`Dev server running at: ${address}`);
+      logSpace();
+    } catch (err: any) {
+      if (err) {
+        if (err.code === 'EADDRINUSE' && retryCount < 10) {
+          retryCount++;
+          logError(`Port ${err.port} is in use, trying ${err.port + 1} instead.`);
+          await start(err.port + 1);
+        } else {
+          console.error(err);
+          process.exit(1);
+        }
+      }
     }
+  }
 
-    console.clear();
-    logSpace();
-    logSuccess(`Dev server running at: ${address}`);
-    logSpace();
-  });
+  start(Number(port));
```

Comment:
I think we should check if `port` is a number before, to prevent any issues.

Reply:
you are right, i will use `parseInt` to parse port string to number.